smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.provide('GoogleSmartCard.ConnectorApp.ChromeApiProvider');
+
+goog.require('GoogleSmartCard.PcscLiteClient.API');
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.ServerRequestHandler');
+goog.require('goog.Disposable');
+goog.require('goog.log');
+goog.require('goog.log.Logger');
+goog.require('goog.messaging.AbstractChannel');
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+const Pcsc = GSC.PcscLiteServerClientsManagement;
+const PcscApi = GSC.PcscLiteClient.API;
+
+/** @type {!goog.log.Logger} */
+const logger = GSC.Logging.getScopedLogger('ConnectorApp.ChromeApiProvider');
+
+/**
+ * Returns the result code for chrome.smartCardProviderPrivate API that
+ * corresponds to the given PCSC error.
+ * @param {!PcscApi.ERROR_CODE} errorCode
+ * @return {!chrome.smartCardProviderPrivate.ResultCode}
+ * @throws {Error} Will throw if unknown error code encountered.
+ */
+function convertErrorCodeToEnum(errorCode) {
+  switch (errorCode) {
+    case PcscApi.SCARD_F_INTERNAL_ERROR:
+      return chrome.smartCardProviderPrivate.ResultCode.INTERNAL_ERROR;
+    // TODO(vkovalova): add other error codes
+    default:
+      throw new Error(`Unknown error code ${errorCode} encountered`);
+  }
+}
+
+/**
+ * This class subscribes to a Chrome event with a callback, specified in the
+ * constructor arguments. Unsubscribes on disposal.
+ */
+class ChromeEventListener extends goog.Disposable {
+  constructor(chromeEvent, callback) {
```

Comment:
nit: Add JSDoc to document the types of the arguments.

Reply:
done